src/main/java/hw6/pages/MetalsAndColorsPage.java

```diff
@@ -0,0 +1,92 @@
+package hw6.pages;
+
+import com.epam.jdi.light.elements.common.UIElement;
+import com.epam.jdi.light.elements.complex.WebList;
+import com.epam.jdi.light.elements.composite.WebPage;
+import com.epam.jdi.light.elements.pageobjects.annotations.FindBy;
+import com.epam.jdi.light.elements.pageobjects.annotations.Url;
+import hw6.domain.MetalAndColor;
+import hw6.domain.enums.Element;
+import hw6.domain.enums.Vegetable;
+import hw6.forms.MetalAndColorForm;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+@Url("metals-colors.html")
+public class MetalsAndColorsPage extends WebPage {
+    @FindBy(css = ".results li")
+    public WebList resultStrings;
+    public MetalAndColorForm form;
+    public MetalAndColorForm getForm() {
+        return form;
+    }
+
+    public void submitForm(MetalAndColor metalAndColor) {
+        for (Integer number : metalAndColor.getSummary()) {
+            MetalAndColorForm.summary.select(String.valueOf(number));
+        }
+        for (Element element : metalAndColor.getElements()) {
+            MetalAndColorForm.elements.select(element);
+        }
+        MetalAndColorForm.metals.select(metalAndColor.getMetals());
+        MetalAndColorForm.colors.select(metalAndColor.getColors());
+        // Reset default dropdown value
+        MetalAndColorForm.vegetables.select(Vegetable.VEGETABLE);
+        for (Vegetable vegetable : metalAndColor.getVegetables()) {
+            MetalAndColorForm.vegetables.select(vegetable);
+        }
+        MetalAndColorForm.submitBtn.click();
+    }
+
+   public void assertResult(MetalAndColor metalAndColor) {
+       for (UIElement resultString : resultStrings) {
```

Comment:
Imagin the situation if **resultStrings** is empty.

Reply:
This approach can be simplified, you can compare only two List<Strings>.

Reply:
-  Have added check for `resultStrings`.

-  Don't really understand how we can utilize two Lists here? Is it connected with usage of Reflection API for the class to compare it fields with actual Strings?